org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/workspace/GradleBuild.java

```diff
@@ -41,42 +32,15 @@
     /**
      * Attempts to synchronize the build with the workspace.
      * <p/>
-     * The synchronization happens asynchronously. In case of a failure, the user will be notified
-     * once after the build has finished.
-     * <p/>
-     * This is equivalent to calling {@code synchronize(NewProjectHandler.NO_OP)}
-     */
-    void synchronize();
-
-    /**
-     * Attempts to synchronize the build with the workspace.
-     * <p/>
-     * The synchronization happens asynchronously. In case of a failure, the user will be notified
-     * once the build has finished.
-     * <p/>
-     * This is equivalent to calling {@code synchronize(newProjectHandler, AsyncHandler.NO_OP)}
+     * The synchronization happens synchronously. In case of a failure, the method throws a
+     * {@link CoreException} which contains the necessary status and error message about the
+     * failure.
      *
      * @param newProjectHandler how to handle newly added projects
+     * @param tokenSource the cancellation token source
```

Comment:
⭕️ Can we get rid of the token source argument and construct it ourselves?

Reply:
Well, you need the tokenSource to be available on the caller side to attach the cancellation for instance to the preview page.